mne/beamformer/_lcmv.py

```diff
@@ -163,24 +227,27 @@ def _apply_lcmv(data, info, tmin, forward, noise_cov, data_cov, reg,
         if not return_single:
             logger.info("Processing epoch : %d" % (i + 1))
 
-        # SSP and whitening
+
+        # SSP and whitening        
         if info['projs']:
             M = np.dot(proj, M)
-        M = np.dot(whitener, M)
+        if noise_cov:
+            M = np.dot(whitener, M)
 
         # project to source space using beamformer weights
-
         if is_free_ori:
             sol = np.dot(W, M)
             logger.info('combining the current components...')
             sol = combine_xyz(sol)
-            sol *= noise_norm_inv[:, None]
+
+            if apply_noise_norm:
+                sol *= noise_norm_inv[:, None]
         else:
             # Linear inverse: do computation here or delayed
             if M.shape[0] < W.shape[0] and pick_ori != 'max-power':
                 sol = (W, M)
             else:
-                sol = np.dot(W, M)
+                sol = np.dot(W, M)         
```

Comment:
run flake8 on your files and add it to your editor to have real time checks of pep8 rules